internal/app_config.go

```diff
@@ -0,0 +1,127 @@
+package internal
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/pelletier/go-toml"
+)
+
+var DEF_CONFIG_ORDER = []string{
```

Comment:
One of the API ctor's allows customer to set custom list of config dirs or files.
So, this won't work if the custom client's list of conf dirs differs from the `DEF_CONFIG_ORDER`. 

It's not critical at the moment, however can become critical as more customers build custom sota client.

Reply:
Sounds like we'll need a configuration knob for this configuration knob. Or - you can probably update the systemd service unit to pass the config dirs you want to the daemon's CLI

Reply:
The dir/file list can be set at a factory level in some file under `/usr`; a default yocto recipe defines it the default list; a customer can override it. Then, both aklite/custom-client and fioconfig reads it from that file.
?

Reply:
maybe - or maybe we just state that this override is only intended for unit tests. production should the default path

Reply:
I guess it makes sense, no value in making things more complex. Anyway those defaults dirs are so generic that they are suitable for all use-cases.

If so, we probably need to add a comment about it to this ctor https://github.com/foundriesio/aktualizr-lite/blob/b2140a50e48537680f5832141141d6db579c6e00/include/aktualizr-lite/api.h#L194.

Or maybe remove it at all and just use `cmdline_args["config"]` and this ctor https://github.com/foundriesio/aktualizr-lite/blob/b2140a50e48537680f5832141141d6db579c6e00/include/aktualizr-lite/api.h#L203 for the unit tests and the custom sota example.

Reply:
I think a comment is probably sufficient. I think Patti may have introduced this line for their tests - so just don't do anything that might accidentally break them

Reply:
It's funny how my comment to fioconfig code returned back to me as an action item over aklite :).